### ISSUE TITLE
[Feature request] Avoid panel + center thumbnails with 100% width/height container

### ISSUE THREAD

**Issue by Hombremaledicto** [passingthru67/workspaces-to-dock#120]
Hi, i'm currently using dash-to-panel with this extension, and the WS list placed on top. However, if i place it at the bottom of the screen, the WS list will appear above the panel:

[![Schermata_del_2017-10-06_17-50-16.png](https://s1.postimg.org/6fu3kt4ehr/Schermata_del_2017-10-06_17-50-16.png)](https://postimg.org/image/8eqab59wt7/)

While on windows, we have this layout:

[![maxresdefault.jpg](https://s1.postimg.org/1q1m0ktd1r/maxresdefault.jpg)](https://postimg.org/image/18bkbzrzgr/)

Having the workspaces list container placed after the panel would solve the issue (in this case, having a horizontal panel and the workspaces-to-dock thumbs on the same screen edge). Also, as you can see, the WS list container on windows takes all the available width, while mantaining the thumbs centered (MacOs does the same with its mission control effect), which i think would be a very nice option to have.

**Comment by passingthru67** [comment]
@Hombremaledicto, Would you repost those images? I'm not sure if I'm seeing the same thing as you. I'm testing using the latest DtP and WtD from master (with advanced centering options). With both placed on the bottom, WtD is below DtP and not above. In fact, they seem to work well together.

**Comment by Hombremaledicto** [comment]
Unfortunately the original pics are lost. Anyway, the current behaviour here:

![screenshot-2018-05-03-18 28 52](https://user-images.githubusercontent.com/7595097/39590048-08aa6aae-4f00-11e8-83ef-3ea66d06fec0.png)
![screenshot-2018-05-03-18 28 56](https://user-images.githubusercontent.com/7595097/39590056-0e696bc0-4f00-11e8-96c5-45c0cfea4383.png)

The proposed one:

![vd_top](https://user-images.githubusercontent.com/7595097/39590089-2cffb86e-4f00-11e8-90fc-627845aecdce.png)

**Comment by passingthru67** [comment]
What verson of WtD are you using? Is it the latest master branch with the advanced center options?

**Comment by Hombremaledicto** [comment]
Yes, master branch, just updated to the latest and reloaded the shell before taking those pictures (DtP also at the latest from master)

**Comment by passingthru67** [comment]
Maybe I'm misunderstanding. Are you proposing to have WtD show up above DtP when it is triggered at the bottom of the screen?

**Comment by Hombremaledicto** [comment]
Yes, sorry in case i explained myself poorly (and by the way, i still have to thank you for implementing the centered thumbs).
I posted the win10 screen since it was doing so - well, not anymore, but that's the concept

**Comment by passingthru67** [comment]
What if I added an option for screen edge margin? The padding would default to the Gnome top panel height.

Basically, when WtD is triggered, it would still show overlapping DtP, but there would be a transparent margin between the dock and the edge of the screen where PtD is placed. This option would also benefit others that would like WtD to appear below the Gnome Shell top panel (when WtD is positioned at the top).

Does that make sense?

**Comment by Hombremaledicto** [comment]
Sort of, yes. The infos within the panel (notifications, date, and others from extensions) will still be visible, even though the panel won't take inputs (if i understood correctly), still a welcomed feature IMO

**Comment by passingthru67** [comment]
A added an option to set padding between the dock and the screen edge (see master branch). Unfortunately, I haven't figured out how to pass mouse events through the dock without making it non-reactive. I'm not sure it's even possible to do with GJS.

**Comment by Hombremaledicto** [comment]
It looks already great:

![screenshot-2018-05-04-19 49 09](https://user-images.githubusercontent.com/7595097/39643209-50d55b70-4fd4-11e8-807c-8c9ccde4ff9c.png)

I had to turn down opacity a little to show dash-to-panel, yet still great.  
I'd like to help on sending mouse events to DtP, but i know nothing on the topic.

**Comment by passingthru67** [comment]
I'll keep investigating GJS for an answer to the mouse clicks. But, this will have to do for now.

**Comment by Hombremaledicto** [comment]
Thank you very much !
